pkg/apiserver/rest/asset_scan_controller.go

```diff
@@ -83,14 +91,26 @@ func (s *ServerImpl) PatchAssetScansAssetScanID(ctx echo.Context, assetScanID mo
 	}
 
 	// check that an asset scan with that id exists.
-	_, err = s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
+	existingAssetScan, err := s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
 	if err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("asset scan was not found. assetScanID=%v: %v", assetScanID, err))
 		}
 		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan. assetScanID=%v: %v", assetScanID, err))
 	}
 
+	// check for valid resource cleanup state transition
+	if resourceCleanupStatus, ok := assetScan.GetResourceCleanupStatus(); ok {
+		existingResourceCleanupStatus, ok := existingAssetScan.GetResourceCleanupStatus()
+		if !ok {
+			return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to retrieve ResourceCleanupStatus for existing asset scan. assetScanID=%v", existingAssetScan.Id))
+		}
+		err = existingResourceCleanupStatus.IsValidTransition(resourceCleanupStatus)
+		if err != nil {
+			return sendError(ctx, http.StatusBadRequest, err.Error())
+		}
+	}
```

Comment:
If we're moving to handling stuff in the rest controllers then shouldn't we handle the LastTransitionTime in the rest controller too? i.e. detect that a status change has been PATCH/PUTed and then update the time stamp to prevent the orchestrator or other API clients from a) having to manage it, b) potentially faking the data.

Reply:
We briefly touched this problem with @adamtagscherer. And the best would be to make the API server to be responsible for updating the `LastTransitionTime` field for `Status` objects. But we thought that it would be better to do the refactor for all the `Status` objects first and than have a separate PR for making the API server setting the `LastTransitionTime`.

Reply:
I think this makes sense, I was also thinking that we might like to add UpdatedAt/CreatedAt fields to all objects in the API.

Reply:
I created an issue for this: https://github.com/openclarity/vmclarity/issues/720